src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -67,108 +54,122 @@ func Setup(version string, buildDate string) *rotator.SizeRotator {
 	// Set log files to 1 MB
-	file.RotationSize = 1 * 1024 * 1024
+	logFile.RotationSize = 1 * 1024 * 1024
 	// Keep up to 20 log files
-	file.MaxRotation = 20
+	logFile.MaxRotation = 20
 
 	// Loggly has its own timestamp so don't bother adding it in message,
 	// moreover, golog always write each line in whole, so we need not to care about line breaks.
-	errorOut := timestamped(NonStopWriter(os.Stderr, file))
+	errorOut = timestamped(io.MultiWriter(os.Stderr, logFile))
+	debugOut = timestamped(io.MultiWriter(os.Stdout, logFile))
+	golog.SetOutputs(errorOut, debugOut)
+}
 
+func Configure(cfg *config.Config, version string, buildDate string) {
 	if logglyToken == "" {
 		log.Debugf("No logglyToken, not sending error logs to Loggly")
-	} else {
-		log.Debugf("Sending error logs to Loggly")
-		logglyClient = &logglyErrorWriter{loggly.New(logglyToken)}
-		errorOut = NonStopWriter(errorOut, logglyClient)
+		return
 	}
 
-	debugOut := timestamped(NonStopWriter(os.Stdout, file))
-	golog.SetOutputs(errorOut, debugOut)
-	return file
-}
-
-func Configure(cfg *config.Config) (err error) {
 	cfgMutex.Lock()
 	defer cfgMutex.Unlock()
 
-	if logglyClient == nil {
-		return fmt.Errorf("No loggly client to configure.")
+	if version == "" {
+		log.Error("No version configured, Loggly won't include version information")
+		return
 	}
 
-	if cfg.Addr == "" {
-		return fmt.Errorf("No proxy address provided.")
+	if buildDate == "" {
+		log.Error("No build date configured, Loggly won't include build date information")
+		return
 	}
 
-	if err = waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second); err != nil {
-		return fmt.Errorf("Proxy never came online at %s: %q", cfg.Addr, err)
-	}
+	if cfg.Addr != lastAddr {
+		lastAddr = cfg.Addr
 
-	var cli *http.Client
-	if cli, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr); err != nil {
-		return fmt.Errorf("Could not create proxy HTTP client %q.", err)
-	}
+		if cfg.Addr == "" {
+			log.Error("No known proxy, won't report to Loggly")
+			removeLoggly()
+			return
+		}
 
-	logglyClient.l.SetHTTPClient(cli)
+		err := waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second)
+		if err != nil {
+			log.Errorf("Proxy never came online at %v, not logging to Loggly", cfg.Addr)
+			removeLoggly()
+			return
+		}
+
+		var client *http.Client
+		client, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr)
+		if err != nil {
+			log.Errorf("Could not create proxied HTTP client, not logging to Loggly: %v", err)
+			removeLoggly()
+			return
+		}
+
+		log.Debugf("Sending error logs to Loggly via proxy at %v", cfg.Addr)
+
+		lang, _ := jibber_jabber.DetectLanguage()
+		logglyWriter := &logglyErrorWriter{
+			lang:            lang,
+			tz:              time.Local.String(),
+			versionToLoggly: fmt.Sprintf("%v (%v)", version, buildDate),
+			client:          loggly.New(logglyToken),
+		}
+		logglyWriter.client.SetHTTPClient(client)
+		addLoggly(logglyWriter)
+	}
+}
 
-	return nil
+func Close() error {
```

Comment:
Instead of returning the logfile so that main can close it, we just expose a Close() function which main can use instead.